validation/validation_efield_STEPS4/rallpack1.py

```diff
@@ -0,0 +1,327 @@
+    traces_sample_STEPS3 = []
+    traces_sample_STEPS3.append(Trace("t", "ms", multi=multi))
+
+    traces_sample_STEPS3.append(
+        Trace(
+            "V zmin",
+            "mV",
+            multi=multi,
+            reduce_ops={
+                "amin": [],
+                "amax": [],
+            },
+        )
+    )
+    traces_sample_STEPS3.append(copy.deepcopy(traces_sample_STEPS3[-1]))
+    traces_sample_STEPS3[-1].name = "V zmax"
+    if db_STEPS3 is not None:
+        for i in traces_sample_STEPS3:
+            i.add_raw_trace("", db_STEPS3[i.name])
+
+    """Create the sample database"""
+    sample_STEPS3 = TraceDB(
+        "STEPS3",
+        traces_sample_STEPS3,
+        folder_path="",  # configuration.path(os.path.join('validation_efield_STEPS4', 'results', 'STEPS3')),
+        clear_raw_traces_cache=True,
+        clear_refined_traces_cache=True,
+        save_raw_traces_cache=False,
+        save_refined_traces_cache=False,
+    )
+
+    """Sample_STEPS4"""
+    multi = 1000
+    traces_sample_STEPS4 = []
+    traces_sample_STEPS4.append(Trace("t", "ms", multi=multi))
+
+    traces_sample_STEPS4.append(
+        Trace(
+            "V zmin",
+            "mV",
+            multi=multi,
+            reduce_ops={
+                "amin": [],
+                "amax": [],
+            },
+        )
+    )
+    traces_sample_STEPS4.append(copy.deepcopy(traces_sample_STEPS4[-1]))
+    traces_sample_STEPS4[-1].name = "V zmax"
+    if db_STEPS4 is not None:
+        for i in traces_sample_STEPS4:
+            i.add_raw_trace("", db_STEPS4[i.name])
+
+    """Create the sample database"""
+    sample_STEPS4 = TraceDB(
+        "STEPS4",
+        traces_sample_STEPS4,
+        folder_path="",  # configuration.path(os.path.join('validation_efield_STEPS4', 'results', 'STEPS4')),
+        clear_raw_traces_cache=True,
+        clear_refined_traces_cache=True,
+        save_raw_traces_cache=False,
+        save_refined_traces_cache=False,
+    )
+
+    """comparator"""
+
+    comp = Comparator(traceDBs=[benchmark_analytic, sample_STEPS3, sample_STEPS4])
+
+    for i in ["V zmin", "V zmax"]:
+        comp.plot(
+            trace_name_b=i,
+            savefig_path=configuration.path(
+                os.path.join("validation_efield_STEPS4", "results", "pics")
+            ),
+            isdiff=False,
+            suffix="",
+            interactive=False,
+        )
+
+    """Compute the mse"""
+    for tDBnames, mse_tests in comp.mse_refactored(normalized=False).items():
+        print(tDBnames)
+        for k, v in sorted(mse_tests.items(), key=lambda k: k[0]):
+            # the std. mesh is quite coarse and the error with the analytic solution may be considered still big.
+            # However, we are also comparing with STEPS 3 where we can be much more strict.
+            err = 1e-1 if "analytic" in tDBnames else 1e-12
+            print(k, *v.items(), f" Max accepted err: {err}")
+
+            assert v["amax"] < err
```

Comment:
Also, is this check ok? When running the validation, I got a failed assert because of `V zmax ('mean', 1369.1667724867286) ('amax', 1369.1667724867286) ('amin', 1369.1667724867286) ('norm_factor', 1.0)  Max accepted err: 0.1`

Reply:
very interesting. Are you use your `multi` is set up as it is for me?

Reply:
I would keep them separate as they are already complex and check_results is very similar to rallpack1.py in postproc. What are the advantages of moving everything in 1 function?

Reply:
In results/pics I have:
![V_zmax_STEPS3_vs_STEPS4](https://user-images.githubusercontent.com/25343549/161908485-aac21034-f5d8-42c0-a381-d422222a637b.jpg)

But the data for V zmax in results/STEPS_4 looks ok (around -65mV).

Not sure what `multi` does but it seems to always be set to `1000`.

Reply:
thank you @jlallouette ! Something strange is happening here. In any case I want to move this test to the CI and in STEPS4ModelRelease. So this test will be removed and this pr will have a much clearer target. Because I want to move stuff around I moved this pr back to draft